sdk/python/kfp/dsl/type_utils.py

```diff
@@ -42,9 +45,9 @@
     'dataset':
```

Comment:
Use io_types for this too?

Reply:
Will do that in a follow up PR, as that change will require changing more test samples.